src/pykeen/models/unimodal/pair_re.py

```diff
@@ -35,34 +38,56 @@ class PairRE(ERModel):
     def __init__(
         self,
         embedding_dim: int = 200,
-        p: int = 2,
-        power_norm: bool = True,
-        entity_initializer: Hint[Initializer] = xavier_normal_,
-        relation_initializer: Hint[Initializer] = xavier_normal_,
+        p: int = 1,
+        power_norm: bool = False,
+        entity_initializer: Hint[Initializer] = uniform_,
+        entity_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        entity_normalizer: Hint[Normalizer] = functional.normalize,
+        entity_normalizer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_initializer: Hint[Initializer] = uniform_,
+        relation_initializer_kwargs: Optional[Mapping[str, Any]] = None,
         **kwargs,
     ) -> None:
         r"""Initialize PairRE via the :class:`pykeen.nn.modules.PairREInteraction` interaction.
 
         :param embedding_dim: The entity embedding dimension $d$. Defaults to 200. Is usually $d \in [50, 300]$.
         :param p: The $l_p$ norm. Defaults to 2.
         :param power_norm: Should the power norm be used? Defaults to true.
-
-        .. warning:: Due to the lack of an official implementations, not all details are known.
         """
+        # The entity representations are normalized to L2 unit length
+        # cf. https://github.com/alipay/KnowledgeGraphEmbeddingsViaPairedRelationVectors_PairRE/blob/0a95bcd54759207984c670af92ceefa19dd248ad/biokg/model.py#L232-L240
+        entity_normalizer_kwargs = entity_normalizer_kwargs or dict()
+        entity_normalizer_kwargs.setdefault("p", 2)
+        entity_normalizer_kwargs.setdefault("dim", -1)
+        # update initializer settings, cf.
+        # https://github.com/alipay/KnowledgeGraphEmbeddingsViaPairedRelationVectors_PairRE/blob/0a95bcd54759207984c670af92ceefa19dd248ad/biokg/model.py#L45-L49
+        # https://github.com/alipay/KnowledgeGraphEmbeddingsViaPairedRelationVectors_PairRE/blob/0a95bcd54759207984c670af92ceefa19dd248ad/biokg/model.py#L29
+        # https://github.com/alipay/KnowledgeGraphEmbeddingsViaPairedRelationVectors_PairRE/blob/0a95bcd54759207984c670af92ceefa19dd248ad/biokg/run.py#L50
+        entity_initializer_kwargs = entity_initializer_kwargs or dict()
+        embedding_range = 14 / embedding_dim
+        entity_initializer_kwargs.setdefault("a", -embedding_range)
```

Comment:
do we like doing the setdefaults or should we have the defaults be class variables so they can be introspected?

Reply:
Introspection sounds better :wink:

Reply:
I can do this for you

Reply:
Wait, these are dynamic. So maybe this isn't possible.

Reply:
maybe like this? 8246328

Reply:
ahh i missed those. Looks good

Reply:
How is this solved in other places? Do we already have some method(s) there?

Reply:
No, there isn't a general solution, but we can keep this code block in mind for later